### PR TITLE
Improved error messages

### PR DESCRIPTION
**Pull request by rfuest**
This PR improves the error messages by adding more information to the `ErrorCode` variants. Some errors didn't previously show the exact cause of the problem, because multiple errors shared a single enum variant (e.g. for multiple pins). `test.log` should give a good overview of the changes.

I decided to rename `Mode1` - `Mode3` to `Simple`, `Complex` and `Registered`. I think these names should be more recognizable for users, because they are also used in the datasheets. Or is there an advantage of having the old names, that I didn't consider?

### CODE REVIEW

File: baseline/GAL16V8_complex_in.jed [hunk 1]
```diff
@@ -22,11 +22,11 @@ Device:         GAL16V8
 *L1824 01110111111111111111111111111111
 *L1856 11111111111111111101111111111111
 *L2048 00000111
-*L2056 0100110101101111011001000110010100110010001000000011101100100000
+*L2056 0100001101101111011011010111000001101100011001010111100000100000
```
Comment (simon-frankau) [inline baseline/GAL16V8_complex_in.jed:25]: Any idea what caused this to change? I would expect this to be a pure refactoring, with no change to the generated .jeds.

**Review by simon-frankau — COMMENTED** [review]
Thanks, I appreciate these changes. Much of galette was following the pattern of galasm pretty directly, as I was learning about both Rust and GALs, so I mostly stuck to the same conventions, but I am very happy to switch to more readable conventions!

File: baseline/GAL16V8_complex_in.jed (hunk 1 above)
Comment (rfuest) [reply baseline/GAL16V8_complex_in.jed:25]: These are the signature bytes and I did change the signature in `GAL16V8_complex_in.pld` from `Mode2` to `Complex`.

Comment (simon-frankau) [reply baseline/GAL16V8_complex_in.jed:25]: D'oh! Obvious in retrospect. Thanks.

**Comment by simon-frankau** [comment]
Thanks!

**Pull request merged**